File: csharp/cars-assemble/CarsAssemble.cs
using System;

static class AssemblyLine
{
    public static double SuccessRate(int speed)
    {
        var successPercentage = speed switch
        {
            0 => 0.0,
            >= 1 and <= 4 => 100.0,
            >= 5 and <= 8 => 90.0,
            9 => 80.0,
            _ => 77.0
        };

        return 0.01 * successPercentage;
    }
    
    public static double ProductionRatePerHour(int speed) =>
         CarsPerHour(speed) * SuccessRate(speed);

    public static int WorkingItemsPerMinute(int speed) =>
        (int)ProductionRatePerHour(speed) / 60;

    private static double CarsPerHour(int speed) =>
        221 * speed;
}

File: fsharp/cars-assemble/Iteration1/CarsAssemble.fs
module CarsAssemble

let carsPerHour = 221

let productionRatePerHour (speed: int): float =
    let basicOutput = float(speed * carsPerHour)

    if (speed >= 1 && speed <= 4) then
        basicOutput
    elif (speed >= 5 && speed <= 8) then
        basicOutput * 0.9
    elif (speed = 9) then
        basicOutput * 0.8
    else
        basicOutput * 0.77

let workingItemsPerMinute(speed: int): int =
    int((productionRatePerHour speed) / 60.0)